### PR TITLE
Add support for reactive in the server extension

### PR DESCRIPTION
**Pull request by carlesarnal**
Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [x] Pull Request contains link to the issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
This PR introduces support for generating reactive code using the server extension
</summary>

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

File: server/integration-tests/reactive/pom.xml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    <name>Quarkus - Openapi Generator - Server - Integration Tests - Resteasy</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-undertow</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive</artifactId>
```
Comment (hbelmiro) [inline server/integration-tests/reactive/pom.xml:21]:
```suggestion
            <artifactId>quarkus-rest</artifactId>
```

File: server/integration-tests/reactive/pom.xml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-openapi-generator-server-integration-tests-parent</artifactId>
+        <groupId>io.quarkiverse.openapi.generator</groupId>
+        <version>3.0.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-openapi-generator-server-integration-tests-resteasy-reactive</artifactId>
```
Comment (hbelmiro) [inline server/integration-tests/reactive/pom.xml:11]:
```suggestion
    <artifactId>quarkus-openapi-generator-server-integration-tests-quarkus-rest</artifactId>
```

File: server/integration-tests/reactive/pom.xml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-undertow</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
```
Comment (hbelmiro) [inline server/integration-tests/reactive/pom.xml:25]:
```suggestion
            <artifactId>quarkus-rest-jackson</artifactId>
```

File: server/integration-tests/resteasy/pom.xml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    <name>Quarkus - Openapi Generator - Server - Integration Tests - Resteasy</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-undertow</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-rest</artifactId>
```
Comment (hbelmiro) [inline server/integration-tests/resteasy/pom.xml:21]:
```suggestion
            <artifactId>quarkus-resteasy</artifactId>
```

File: server/integration-tests/resteasy/pom.xml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-rest-jackson</artifactId>
```
Comment (hbelmiro) [inline server/integration-tests/resteasy/pom.xml:25]:
```suggestion
            <artifactId>quarkus-resteasy-jackson</artifactId>
```

**Review by hbelmiro — CHANGES_REQUESTED** [review]
Thank you @carlesarnal.

It seems like you're confusing the new dependencies for the classic and reactive implementations.
See https://quarkus.io/guides/rest-migration#dependencies.

RESTEasy Classic is still known as RESTEasy (quarkus-resteasy)
RESTEasy Reactive now is Quarkus REST (quarkus-rest).

I suggested changes to the `pom.xml` files.

-----

I also suggest using Maven Profiles to avoid test duplication. With that, you can have just one module and set a flag to run them with each profile (classic or reactive).
[We do that in the client module](https://github.com/quarkiverse/quarkus-openapi-generator/blob/7a255ddc778661e1010d0a679ebbe24f735e2a73/client/integration-tests/pom.xml#L93).

That's all you need since all the tests are identical for both implementations.
In case we need specific tests for each implementation in the future, we can use the `org.junit.jupiter.api.Tag`, like we did [here](https://github.com/quarkiverse/quarkus-openapi-generator/blob/7a255ddc778661e1010d0a679ebbe24f735e2a73/client/integration-tests/config-key/src/test/java/io/quarkiverse/openapi/generator/configkey/QuarkusConfigKeyOpenApiTest.java#L18).

Our CI already handles both profiles. See, for example, [the job for the reactive profile](https://github.com/quarkiverse/quarkus-openapi-generator/blob/7a255ddc778661e1010d0a679ebbe24f735e2a73/.github/workflows/build.yml#L75).

**Comment by carlesarnal** [comment]
> Thank you @carlesarnal.
> 
> It seems like you're confusing the new dependencies for the classic and reactive implementations. See https://quarkus.io/guides/rest-migration#dependencies.
> 
> RESTEasy Classic is still known as RESTEasy (quarkus-resteasy) RESTEasy Reactive now is Quarkus REST (quarkus-rest).
> 
> I suggested changes to the `pom.xml` files.
> 
> I also suggest using Maven Profiles to avoid test duplication. With that, you can have just one module and set a flag to run them with each profile (classic or reactive). [We do that in the client module](https://github.com/quarkiverse/quarkus-openapi-generator/blob/7a255ddc778661e1010d0a679ebbe24f735e2a73/client/integration-tests/pom.xml#L93).
> 
> That's all you need since all the tests are identical for both implementations. In case we need specific tests for each implementation in the future, we can use the `org.junit.jupiter.api.Tag`, like we did [here](https://github.com/quarkiverse/quarkus-openapi-generator/blob/7a255ddc778661e1010d0a679ebbe24f735e2a73/client/integration-tests/config-key/src/test/java/io/quarkiverse/openapi/generator/configkey/QuarkusConfigKeyOpenApiTest.java#L18).
> 
> Our CI already handles both profiles. See, for example, [the job for the reactive profile](https://github.com/quarkiverse/quarkus-openapi-generator/blob/7a255ddc778661e1010d0a679ebbe24f735e2a73/.github/workflows/build.yml#L75).

Tests are not exactly identical (the test class is). There are two different implementations in the Java projects one [reactive](https://github.com/quarkiverse/quarkus-openapi-generator/pull/731/files#diff-e47e71d7fe34b0209c2b2748505c2136ce29c0c92e850493527ac592e57b4b61) and another [non-reactive](https://github.com/quarkiverse/quarkus-openapi-generator/pull/731/files#diff-9049dfd83f1a69611f7d7113500e744ef2f9409b00d842d3a4643e117ac68861), and, if I merge them both into a single module, it will fail. I can make the tests use profiles to align with the rest of the project, but I don't think it's feasible to merge both modules.

**Review by hbelmiro — APPROVED** [review]
@carlesarnal I see. I don't see a benefit in using profiles then.

**Pull request merged**